src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -197,12 +203,17 @@ <R extends AstNode> R acceptAndYield(SqlAstNode node, AstVisitorValueDescriptor<
     // Table Mutation: insertion
 
     @Override
-    public void visitStandardTableInsert(TableInsertStandard tableInsertStandard) {
-        var tableName = tableInsertStandard.getTableName();
-        var astElements = new ArrayList<AstElement>(tableInsertStandard.getNumberOfValueBindings());
-        for (var columnValueBinding : tableInsertStandard.getValueBindings()) {
-            var astValue = acceptAndYield(columnValueBinding.getValueExpression(), FIELD_VALUE);
+    public void visitStandardTableInsert(TableInsertStandard tableInsert) {
+        var tableName = tableInsert.getTableName();
+        var astElements = new ArrayList<AstElement>(tableInsert.getNumberOfValueBindings());
+        for (var columnValueBinding : tableInsert.getValueBindings()) {
             var columnExpression = columnValueBinding.getColumnReference().getColumnExpression();
+
+            if (columnValueBinding.getValueExpression() == null) {
+                throw new FeatureNotSupportedException();
+            }
+            var astValue = acceptAndYield(columnValueBinding.getValueExpression(), FIELD_VALUE);
```

Comment:
Let's rewrite the same as proposed below. This way it is obvious that `valueExpression` can't be `null` past the `if` block, and also that we pass the same value we have just checked for `null` to `acceptAndYield`.

```suggestion
            var valueExpression = columnValueBinding.getValueExpression();
            if (valueExpression == null) {
                throw new FeatureNotSupportedException();
            }
            var astValue = acceptAndYield(valueExpression, FIELD_VALUE);
```